keps/0019-package-api-versioning.md

```diff
@@ -175,21 +176,25 @@ Two proposals: singular and composite package/operator version. In the case of p
 
   Example:
 
-  | Time | Package/operator version | App version        | Comment                                            |
-  | ---- | ---------------          | ------------------ | -------------------------------------------------- |
-  | T0   | 1.0.0                    | 2.3.0              | Initial release based on app version 2.3.x         |
-  | T1   | 1.1.0                    | 3.0.0              | Release based on app version 3.0.x                 |
-  | T2   | 1.2.0                    | 2.3.0              | New operator-related feature for app version 2.3.x |
-  | T2   | 1.3.0                    | 3.0.0              | New operator-related feature for app version 3.0.x |
-  | T3   | 1.4.0                    | 2.3.1              | Bug fix release for app version 2.3.x              |
-  | T4   | 1.5.0                    | 2.3.1              | Important bug fix A in operator-related code       |
-  | T4   | 1.6.0                    | 3.0.0              | Important bug fix A in operator-related code       |
-  | T5   | 1.6.1                    | 2.3.1              | Small bug fix B in operator-related code           |
-  | T5   | 1.6.2                    | 3.0.0              | Small bug fix B in operator-related code           |
-
-  The example above illustrates two sources of confusion:
-  - A higher package version can contain a lower app version
-  - There's no relation between the package version growth and the app version growth
+  | Time | Version  | App version        | Comment                                                     |
+  | ---- | -------- | ------------------ | ----------------------------------------------------------- |
+  | T0   | 1.0.0    | 2.3.0              | Initial release based on app version 2.3.x                  |
+  | T1   | 2.0.0    | 3.0.0              | Release based on app version 3.0.x                          |
+  | T2   | 2.1.0    | 3.0.0              | New operator-related feature A for latest app version 3.0.x |
+  | T2   | 1.1.0    | 2.3.0              | Back-port feature A for app version 2.3.x on demand         |
+  | T3   | 1.1.1    | 2.3.1              | Bug fix release for app version 2.3.x                       |
+  | T4   | 2.2.0    | 3.0.0              | Important bug fix B in operator code for latest 3.0.x       |
+  | T4   | 1.2.0    | 2.3.1              | Back-port bug fix B in operator code for 2.3.x on demand    |
+  | T5   | 2.2.1    | 3.0.0              | Small bug fix C in operator code for latest 3.0.x           |
+  | T5   | 1.2.1    | 2.3.1              | Back-port bug fix C in operator code for 2.3.x on demand    |
```

Comment:
What happens when a breaking change needs to happen the `1.x` version family? Does it bump to the next available major version, which in this simplified scenario would be `3.x`?

Reply:
It looks like the part of the version that keeps track of what is in the operator wrt operator-related features is being manually synced between version families? e.g., `1.2.1` has the same operator-related features and bug fixes as `2.2.1` because they're both `x.2.1`?

Reply:
when we had done GA an operator are we allowed to introduce breaking changes? It will need a reset of major version ofc. And versioning will be just one thing, as we would definitively need to provide an upgrade path for people already running a "GA" version. 
And it will be always done with supported versions in mind. 

In case we just reset Kafka from 0. As its a breaking change. We support two versions, so we will have to introduce two major versions, explaining the upgrade path to them also.